external-crates/move/move-vm/runtime/src/session.rs

```diff
@@ -99,6 +102,7 @@ impl<'r, 'l, S: MoveResolver> Session<'r, 'l, S> {
         gas_meter: &mut impl GasMeter,
     ) -> VMResult<SerializedReturnValues> {
         let bypass_declared_entry_check = true;
+        let gas_rem = gas_meter.remaining_gas().into();
```

Comment:
This and the new function for gas profiling are very similar, so what would you think of this function taking an `Option<&mut GasProfiler` and then doing an `unwrap_or_else`?